include/zephyr/drivers/i3c.h

```diff
@@ -0,0 +1,1705 @@
+	{								\
+		.bus = DEVICE_DT_GET(DT_BUS(node_id)),			\
+		.static_addr = DT_PROP_BY_IDX(node_id, reg, 0),		\
+		.pid = ((uint64_t)DT_PROP_BY_IDX(node_id, reg, 1) << 32)\
+		       | DT_PROP_BY_IDX(node_id, reg, 2),		\
+		.ibi_cb = ibi_cb,					\
+		.init_dynamic_addr =					\
+			DT_PROP_OR(node_id, assigned_address, 0),	\
+	}
+
+/**
+ * @brief Structure initializer for i3c_device_desc from devicetree
+ *
+ * This helper macro expands to a static initializer for a <tt>struct
+ * i3c_device_desc</tt> by reading the relevant bus and device data
+ * from the devicetree.
+ *
+ * @param node_id Devicetree node identifier for the I3C device whose
+ *                struct i3c_device_desc to create an initializer for
+ */
+#define I3C_DEVICE_DESC_DT(node_id)					\
+	{								\
+		.bus = DEVICE_DT_GET(DT_BUS(node_id)),			\
+		.static_addr = DT_PROP_BY_IDX(node_id, reg, 0),		\
+		.pid = ((uint64_t)DT_PROP_BY_IDX(node_id, reg, 1) << 32)\
+		       | DT_PROP_BY_IDX(node_id, reg, 2),		\
+		.init_dynamic_addr =					\
+			DT_PROP_OR(node_id, assigned_address, 0),	\
+	}
+
+/**
+ * @brief Structure initializer for i3c_device_desc from devicetree instance
+ *
+ * This is equivalent to
+ * <tt>I3C_DEVICE_DESC_DT_WITH_IBI_PAYLOAD(DT_DRV_INST(inst), ibi_cb)</tt>.
+ *
+ * @param inst Devicetree instance number
+ * @param ibi_cb Pointer to IBI callback.
+ */
+#define I3C_DEVICE_DESC_DT_INST_WITH_IBI_CB(inst, ibi_cb)		\
+	I3C_DEVICE_DESC_DT_WITH_IBI_CB(DT_DRV_INST(inst, ibi_cb))
+
+/**
+ * @brief Structure initializer for i3c_device_desc from devicetree instance
+ *
+ * This is equivalent to
+ * <tt>I3C_DEVICE_DESC_DT(DT_DRV_INST(inst))</tt>.
+ *
+ * @param inst Devicetree instance number
+ */
+#define I3C_DEVICE_DESC_DT_INST(inst) \
+	I3C_DEVICE_DESC_DT(DT_DRV_INST(inst))
+
+/**
+ * @brief Like DEVICE_DT_DEFINE() with I3C target device specifics.
+ *
+ * Defines a I3C target device which implements the I3C target device API.
+ *
+ * @param node_id The devicetree node identifier.
+ *
+ * @param init_fn Name of the init function of the driver.
+ *
+ * @param pm_device PM device resources reference (NULL if device does not use PM).
+ *
+ * @param data_ptr Pointer to the device's private data.
+ *
+ * @param cfg_ptr The address to the structure containing the
+ *                configuration information for this instance of the driver.
+ *
+ * @param level The initialization level. See SYS_INIT() for
+ *              details.
+ *
+ * @param prio Priority within the selected initialization level. See
+ *             SYS_INIT() for details.
+ *
+ * @param api_ptr Provides an initial pointer to the API function struct
+ *                used by the driver. Can be NULL.
+ */
+#define I3C_DEVICE_DT_DEFINE(node_id, init_fn, pm_device,		\
+			     data_ptr, cfg_ptr, level, prio,		\
+			     api_ptr, ...)				\
+	DEVICE_DT_DEFINE(node_id, init_fn, pm_device,			\
+			 data_ptr, cfg_ptr, level, prio,		\
+			 api_ptr, __VA_ARGS__)
```

Comment:
what's the reason for this _wrapper_?

Reply:
It originally came from I2C. I wish to keep the wrapper here so that it can be extended to incorporate statistics tracking later.